randovania/cli/database.py

```diff
@@ -297,25 +296,31 @@ def list_paths_with_dangerous_logic(args):
     print_only_area = args.print_only_area
     count = 0
 
+    from randovania.game_description.db.node import NodeContext
+    from randovania.game_description.resources.resource_collection import ResourceCollection
+
+    context = NodeContext(None, ResourceCollection(), game.resource_database, game.region_list)
```

Comment:
There are several places where it uses `None` for `GamePatches`.
In  the tests there is even one explicity type cast for `None` to be `GamePatches`.
If it's so common to provide `None` it should be reflected in the type annotation for the field in class `NodeContext`.

Reply:
I know. There's just a giant API issue that I'm improvising around until I can figure out a good solution.
GamePatches can't be None at all. Dock Nodes require it to be not None or it'll just fail.
Also I don't want the asserts because a lot of that code is hot path, so it'd be back to casts in some places.

So I'm just improvising in the places that don't have one and just can't get one, sadly.

(Trying to fix this is part of what I'm hoping the split game description thingy can also help with...)

Reply:
> GamePatches can't be None at all.

At least at some places your code proves the opposite. 😛
If assertions are too time consuming, could still add a union type with `| None` to the field and add a FIXME on top of it mentioning the API issues.